src/pybind/mgr/dashboard/frontend/src/app/shared/services/refresh-interval.service.ts

```diff
@@ -0,0 +1,36 @@
+import { Injectable } from '@angular/core';
+
+import { BehaviorSubject } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class RefreshIntervalService {
+  // Observable sources
+  private intervalDataSource = new BehaviorSubject(null);
+
+  // Observable streams
+  intervalData$ = this.intervalDataSource.asObservable();
+
+  private intervalTime: number;
+  intervalID: number;
```

Comment:
I think this can be made private, too. Or is there any need to make this available externally?

Reply:
This ID used in refresh service testing, which I use it to clear the window interval function after the test passed.